python/arcticdb/version_store/_normalization.py

```diff
@@ -937,6 +938,31 @@ def _packb(*args, **kwargs):
     return packb(*args, **kwargs)
 
 
+def allocate_buffer(max_size: int , attempts: int = 4, divisor: int = 2):
+    """Allocate a buffer of max_size, retrying up to attempts times on out-of-memory errors, reducing the size by divisor each time."""
+    assert max_size > 0
+    assert attempts > 0
+    assert divisor > 0
```

Comment:
https://github.com/man-group/ArcticDB/blob/master/python/arcticdb/preconditions.py#L11
Generally use this instead of `assert`